src/Tarjan.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Graphp\Algorithms;
+
+use Fhaculty\Graph\Exception\InvalidArgumentException;
+use Fhaculty\Graph\Graph;
+use Fhaculty\Graph\Set\Vertices;
+use Fhaculty\Graph\Vertex;
+
+/**
+ * @see http://github.com/Trismegiste/Mondrian/blob/master/Graph/Tarjan.php
+ */
+class Tarjan extends BaseGraph
+{
+    private $stack;
+    private $index;
+    private $partition;
+
+    private $indexMap = array();
+    private $lowLinkMap = array();
+
+    /**
+     * @param Graph $graph
+     */
+    public function __construct(Graph $graph)
+    {
+        parent::__construct($graph);
+        $this->indexMap = new \SplObjectStorage();
+        $this->lowLinkMap = new \SplObjectStorage();
+    }
+
+    /**
+     * Get the strongly connected components of this digraph by
+     * the Tarjan algorithm.
+     *
+     * Starting from :
+     * http://en.wikipedia.org/wiki/Tarjan%27s_strongly_connected_components_algorithm
+     *
+     * Corrected with the help :
+     * https://code.google.com/p/jbpt/source/browse/trunk/jbpt-core/src/main/java/org/jbpt/algo/graph/StronglyConnectedComponents.java
+     * @throws \Fhaculty\Graph\Exception\InvalidArgumentException
+     * @return array the partition of this graph : an array of an array of vertices
+     */
+    public function getStronglyConnected()
+    {
+
+        // check is directed
+        $directed = new Directed($this->graph);
+        if ($directed->hasUndirected()) {
+            throw new InvalidArgumentException('Graph shall be directed');
+        }
+
+        $this->stack = array();
+        $this->index = 0;
+        $this->partition = array();
```

Comment:
You're right, but since we were working on a single Graph (passed in the ctor), I was thinking of this class as fixed input. I reworked this part as requested, there is no more ctor, you pass directly the graph instance to the main method.